PlayerMovement.cs

```diff
@@ -1,180 +1,166 @@
+                // slide boost forward
+                rigidBody.AddForce(orientation.transform.forward * slideForce);
             }
         }
     }
 
-    private void StopCrouch() {
+    void StopCrouch() {
+        crouching = false;
+        // reset scale
         transform.localScale = playerScale;
+        // move down
         transform.position = new Vector3(transform.position.x, transform.position.y + 0.5f, transform.position.z);
     }
 
-    private void Movement() {
-        //Extra gravity
-        rb.AddForce(Vector3.down * Time.deltaTime * 10);
-        
-        //Find actual velocity relative to where player is looking
-        Vector2 mag = FindVelRelativeToLook();
-        float xMag = mag.x, yMag = mag.y;
+    void Movement() {
+        // Extra gravity
+        rigidBody.AddForce(Vector3.down * Time.deltaTime * 10);
 
-        //Counteract sliding and sloppy movement
-        CounterMovement(x, y, mag);
-        
-        //If holding jump && ready to jump, then jump
-        if (readyToJump && jumping) Jump();
+        ApplyFriction();
 
         //Set max speed
         float maxSpeed = this.maxSpeed;
-        
+
         //If sliding down a ramp, add force down so player stays grounded and also builds speed
-        if (crouching && grounded && readyToJump) {
-            rb.AddForce(Vector3.down * Time.deltaTime * 3000);
+        if (crouching && grounded) {
+            rigidBody.AddForce(Vector3.down * Time.deltaTime * 3000);
             return;
         }
-        
-        //If speed is larger than maxspeed, cancel out the input so you don't go over max speed
-        if (x > 0 && xMag > maxSpeed) x = 0;
-        if (x < 0 && xMag < -maxSpeed) x = 0;
-        if (y > 0 && yMag > maxSpeed) y = 0;
-        if (y < 0 && yMag < -maxSpeed) y = 0;
 
         //Some multipliers
         float multiplier = 1f, multiplierV = 1f;
-        
+
         // Movement in air
         if (!grounded) {
             multiplier = 0.5f;
             multiplierV = 0.5f;
         }
-        
+
         // Movement while sliding
         if (grounded && crouching) multiplierV = 0f;
 
         //Apply forces to move player
-        rb.AddForce(orientation.transform.forward * y * moveSpeed * Time.deltaTime * multiplier * multiplierV);
-        rb.AddForce(orientation.transform.right * x * moveSpeed * Time.deltaTime * multiplier);
-    }
+        // more easily adjust left/right while in the air than forward/back
+        rigidBody.AddForce(orientation.transform.right * inputDirection.x * moveSpeed * Time.deltaTime * multiplier);
+        rigidBody.AddForce(orientation.transform.forward * inputDirection.y * moveSpeed * Time.deltaTime * multiplier * multiplierV);
 
-    private void Jump() {
-        if (grounded && readyToJump) {
-            readyToJump = false;
+        if (rigidBody.velocity.magnitude > maxSpeed) {
+            rigidBody.velocity = rigidBody.velocity.normalized * maxSpeed;
```

Comment:
Here we can keep the users speed from going over max speed without affecting input

Reply:
This is absolutely horrible and definitely should be avoided!
Imagine that the users slides down a slope, his speed will be capped to his max speed and not the physics max speed.
Its much better to have a ~0.1 or less speed fluctuation, which is barely noticable.